src/RA_Dlg_MemBookmark.cpp

```diff
@@ -774,50 +776,47 @@ void Dlg_MemBookmark::ExportJSON()
 
 void Dlg_MemBookmark::ImportFromFile(std::string sFilename)
 {
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+    std::ifstream ifile{ sFilename };
+
+    Document doc;
+    IStreamWrapper isw{ ifile };
+
+    if (doc.ParseStream(isw); !doc.HasParseError())
     {
-        Document doc;
-        doc.ParseStream(FileStream(pFile));
-        if (!doc.HasParseError())
+        if (doc.HasMember("Bookmarks"))
         {
-            if (doc.HasMember("Bookmarks"))
+            ClearAllBookmarks();
+
+            const auto& BookmarksData = doc["Bookmarks"];
+            for (auto& i : BookmarksData.GetArray())
             {
-                ClearAllBookmarks();
+                wchar_t buffer[256];
+                swprintf_s(buffer, 256, L"%s", Widen(i["Description"].GetString()).c_str());
 
-                const Value& BookmarksData = doc["Bookmarks"];
-                for (SizeType i = 0; i < BookmarksData.Size(); ++i)
-                {
-                    MemBookmark* NewBookmark = new MemBookmark();
+                auto NewBookmark = std::make_unique<MemBookmark>();
+                NewBookmark->SetDescription(buffer);
```

Comment:
Originally did not use delete with new, replaced it with `unique_ptr`. `unique_ptr` follows RAII exactly so it destroys the pointer it owns when it goes out of scope. Unlike CComPtr `release `does not deallocate, it just releases ownership. Its much better to use smart pointers because they have many checks to let you know about any potential problems at compile time (the standard ones at least). Raw pointers just makes apps fail silently or hang in release mode most of the time.

Reply:
I don't think this is going to work without additional changes. `AddBookmark` and `AddBookmarkMap` both store the raw pointer, which will be deallocated when `NewBookmark` goes out of scope.

`m_vBookmarks` should be a vector of `unique_ptr` - it's the authoritative collection of bookmarks. `m_BookmarkMap` is just a convenience mapper for accessing items in `m_vBookmarks`. In my opinion, the number of bookmarks will never be large enough to require a convenience mapper.

Reply:
I would have to disagree, the pointer lives long enough to be inserted in both collections and is deallocated when no longer needed. 
For now I'll put it back and use delete and probably refactor this class entirely in a different PR. While debugging it gives nothing but garbage. I don't think the mapper is needed either.
I can make it work without it being a pointer, but some other work would have to be done to ensure uniqueness.